source/specify-a-query.txt

```diff
@@ -0,0 +1,238 @@
+array field. The following is a list of available array operators:
+
+- ``$all``, which returns documents with arrays that contain all elements in the query
+- ``$elemMatch``, which returns documents if an element in their array field matches all conditions in the query
+- ``$size``, which returns all documents with arrays of a specified size
+
+To learn more about array operators, see the :manual:`Array Query Operators
+</reference/operator/query-array/>` guide in the MongoDB Server manual.
+
+The following example specifies an array operator in a query filter as a
+parameter to the ``find()`` method. The code returns all documents with a
+``type`` array field containing ``2`` elements:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/specify-query/specify-queries.py
+      :start-after: start-find-array
+      :end-before: end-find-array
+      :language: python
+
+   .. output::
+
+      {'_id': 1, 'name': 'apples', 'qty': 5, 'rating': 3, 'color': 'red', 'type': ['fuji', 'honeycrisp']}
+      {'_id': 3, 'name': 'oranges', 'qty': 6, 'rating': 2, 'type': ['naval', 'mandarin']}
+
+Element Operators
+-----------------
+
+Element operators query data based on the presence or type of a field.
+
+To learn more about element operators, see the :manual:`Element Query Operators
+</reference/operator/query-element/>` guide in the MongoDB Server manual.
+
+The following example specifies an element operator in a query filter as a
+parameter to the ``find()`` method. The code returns all documents that have a
+``color`` field:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/specify-query/specify-queries.py
+      :start-after: start-find-element
+      :end-before: end-find-element
+      :language: python
+
+   .. output::
+
+      {'_id': 1, 'name': 'apples', 'qty': 5, 'rating': 3, 'color': 'red', 'type': ['fuji', 'honeycrisp']}
+      {'_id': 2, 'name': 'bananas', 'qty': 7, 'rating': 4, 'color': 'yellow', 'type': ['cavendish']}
+      {'_id': 4, 'name': 'pineapple', 'qty': 3, 'rating': 5, 'color': 'yellow'}
+
+Evaluation Operators
+--------------------
+
+Evaluation operators return data based on evaluations of either individual
+fields or the entire collection's documents.
+
+The following is a list of common evaluation operators:
+
+- ``$text``, which performs a text search on the documents
+- ``$regex``, which returns documents that match a specified regular expression
+- ``$mod``, which performs a modulo operation on the value of a field and
+  returns documents where the remainder is a specified value
+
+To view a full list of evaluation operators, see the :manual:`Evaluation Query Operators
+</reference/operator/query-evaluation/>` guide in the MongoDB Server manual.
+
+The following example specifies an evaluation operator in a query filter as a
+parameter to the ``find()`` method. The code returns all documents with a
+``name`` field that matches the regular expression ``/apple/``:
```

Comment:
How come you're using `r"apple"` instead of `/apple/`? The latter looks more like a regex. 

I like the following example: `list(clxn.find({ "name" : { "$regex" : "p{2,}" }} ))` which will give you any match that has at least 2 consequence p characters.

Reply:
`r"apple` was just the way it worked when I was testing the queries, but I do like yours better as it's probably a more realistic usage of regex, but still simple enough to understand. Updated